frontend/utils/addIdentifier.ts

```diff
@@ -0,0 +1,18 @@
+import type { ApiPagingResponse } from '~/types/api/common'
+/**
+ * Sometimes the api does not provide a `unique id` for objects in an array.
+ * But in `frontend` there are `components` that need  to differentiate between the items.
+ * So an identifier for each item is constructed out of a combination of the `object`.
+ */
+export function addIdentifier<T>(response: ApiPagingResponse<T> | undefined, ...keys: (keyof T)[]) {
+  if (!response) {
+    return
+  }
+  return {
+    data: response.data.map(row => ({
+      ...row,
+      identifier: keys.map(key => row[key]).join('-'),
+    })),
+    paging: response.paging,
+  }
+}
```

Comment:
np: this should work too and is a bit cleaner imo
```suggestion
export function addIdentifier<T>(
  pagingResponse: ApiPagingResponse<T> | undefined,
  keys: (keyof T)[],
) {
  if (!pagingResponse) {
    return
  }
  return {
    ...pagingResponse,
    data: pagingResponse.data
      .map(item => ({
        ...item,
        identifier: keys.map(key => item[key]).join('-'),
      })),
  }
}
```

Reply:
But again, imo this is a `code smell` and could potentially introduce `performance issues` in large data sets.
As this will be done for every row in a table.

I still think `data manipulation` should happen in `backend`.